src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
+const FULL_HEALTH = 100;
+const DAMAGE = 50;
+
 class Animal {
-  // write your code here
+  constructor(name = '') {
+    this.health = FULL_HEALTH;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name = '') {
+    super(name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name = '') {
+    super(name);
+  }
+  bite(production) {
```

Comment:
What does you mean by using `production`? 
You can choose more predictable name: aim, victim etc.